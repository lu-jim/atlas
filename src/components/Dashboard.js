import { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { getStats } from '../redux/dashboard/dashboard';
import ContinentCard from './ContinentCard';

const Dashboard = () => {
  const dispatch = useDispatch();
  const statList = useSelector((state) => state.dashboard.dashboard);
  useEffect(() => {
    if (!statList.length) dispatch(getStats());
  }, []);
  const stats = { All: [0, 0] }; // [Area, Population]
  statList.forEach((country) => {
    if (Object.keys(stats).indexOf(country.region) === -1) {
      stats[country.region] = [country.area + 0, country.population + 0];
    } else {
      stats[country.region][0] += country.area + 0;
      stats[country.region][1] += country.population + 0;
    }
    stats.All[0] += country.area + 0;
    stats.All[1] += country.population + 0;
  });
  const continentList = Object.keys(stats);
  console.log(stats);
  const list = continentList.map((continent) => (
    <ContinentCard
      key={stats[continent][0] + stats[continent][1]}
      continent={continent}
      area={stats[continent][0]}
      population={stats[continent][1]}
    />
  ));
  return (
    <div id="stat-list" className="grid grid-cols-2 md:grid-cols-4 gap-4 bg-sky-900">
      {list || ''}
    </div>
  );
};
export default Dashboard;
