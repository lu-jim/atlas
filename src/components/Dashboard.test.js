import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import TestRenderer from 'react-test-renderer';
import store from '../redux/configureStore';
import Dashboard from './Dashboard';

describe('App Rendering', () => {
  test('Render dashboard component', () => {
    render(
      <BrowserRouter>
        <Provider store={store}>
          <Dashboard />
        </Provider>
      </BrowserRouter>,
    );
    screen.debug();
  });
  test('Dashboard data loads properly', () => {
    const tree = TestRenderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <Dashboard />
        </Provider>
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
