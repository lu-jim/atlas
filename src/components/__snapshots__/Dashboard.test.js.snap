// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Rendering Dashboard data loads properly 1`] = `
<div
  className="grid grid-cols-2 md:grid-cols-3 bg-sky-900"
  id="dashboard"
>
  <a
    className="bg-sky-900 col-span-2 md:col-span-3 bg-sky-900 flex  flex-col text-white py-4 px-2 flex justify-center"
    href="/continent/All"
    id={0}
    onClick={[Function]}
  >
    <h2
      className="text-3xl lg:text-4xl uppercase font-bold text-white"
    >
      All
    </h2>
    <p
      className="text-sm md:text-base pt-2"
    >
       
      Area: 
      0
       m²
    </p>
    <p
      className="text-sm md:text-base"
    >
      Population: 
      0
    </p>
  </a>
</div>
`;
